### PR TITLE
Adding form parsing method to bq connector

### PR DESCRIPTION
**Pull request by mservidio**

### CODE REVIEW

File: bq-connector/docai_bq_connector/connector/BqDocumentMapper.py [hunk 1]
```diff
@@ -59,7 +65,13 @@ def __init__(
         self.dictionary = self._map_document_to_bigquery_schema(self.fields, bq_schema)
 
     def _parse_document(self) -> List[DocumentField]:
-        row = self._parse_entities(self.processed_document.document.entities)
+        row: DocumentRow
+        if self.parsing_methodology == PARSING_METHOD_ENTITIES:
+            row = self._parse_entities(self.processed_document.document.entities)
+        elif self.parsing_methodology == PARSING_METHOD_FORM:
+            row = self._parse_form_entities(self.processed_document.document)
+        else:
```
Comment (holtskinner) [inline bq-connector/docai_bq_connector/connector/BqDocumentMapper.py:73]: Shouldn't you also have a case for `PARSING_METHOD_NORMALIZED_VALUES` ?

File: bq-connector/docai_bq_connector/doc_ai_processing/Processor.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def _process_sync(
         results_json = documentai.types.Document.to_json(results.document)
```
Comment (holtskinner) [inline bq-connector/docai_bq_connector/doc_ai_processing/Processor.py:136]: If you need this as a dictionary, there's a built-in function for it

`documentai.Document.to_dict()`

File: bq-connector/docai_bq_connector/doc_ai_processing/Processor.py [hunk 3]
```diff
@@ -136,7 +136,7 @@ def _process_sync(
...
         return ProcessedDocument(
             document=results.document,
-            dictionary=results_json,
+            dictionary=json.loads(results_json),
```
Comment (holtskinner) [inline bq-connector/docai_bq_connector/doc_ai_processing/Processor.py:139]: If you have it as a dict, you won't need `json.loads()`

File: bq-connector/main.py [hunk 4]
```diff
@@ -262,11 +263,12 @@ def main():
...
 
     processed_doc = connector.run()
-    print(
-        f"Finished processing document - Extracted {len(processed_doc.document.entities)} entities "
-        "and saved results to BigQuery"
-        ""
-    )  # noqa: E127
+    if isinstance(processed_doc, ProcessedDocument):
+        print(
+            f"Finished processing document - Extracted {processed_doc.field_count} fields "
```
Comment (holtskinner) [inline bq-connector/main.py:268]: Maybe change this to show the parsing methodology instead of just "fields"

**Review by holtskinner — CHANGES_REQUESTED** [review]

File: bq-connector/docai_bq_connector/connector/BqDocumentMapper.py (hunk 1 above)
Comment (mservidio) [reply bq-connector/docai_bq_connector/connector/BqDocumentMapper.py:73]: PARSING_METHOD_NORMALIZED_VALUES isn't fully implemented yet. I haven't come across a document yet between the two use-cases I ran that have those fields populated. I can remove it, or we can complete it once we have a better working example.

File: bq-connector/docai_bq_connector/doc_ai_processing/Processor.py [hunk 5]
```diff
@@ -253,15 +253,17 @@ def _process_async(self) -> Union[DocumentOperation, ProcessedDocument]:
             hitl_op_id = hitl_op_full_path.split("/").pop()
 
         return ProcessedDocument(
-            document=document, dictionary=blob_as_bytes, hitl_operation_id=hitl_op_id
+            document=document,
+            dictionary=json.loads(blob_as_bytes),
```
[the remark this answers is not in the document]
Comment (mservidio) [reply bq-connector/docai_bq_connector/doc_ai_processing/Processor.py:257]: This one is reading directly from the file contents, so json.loads may be the least costly.

**Review by holtskinner — APPROVED** [review]

**Pull request merged**